src/arch/x86/reg/amd64_core.rs

```diff
@@ -0,0 +1,155 @@
+use crate::arch::Registers;
+use core::convert::TryInto;
+
+/// 64-bit x86 core registers.
+///
+/// Source: https://github.com/bminor/binutils-gdb/blob/master/gdb/features/arm/arm-core.xml
+#[derive(Default)]
+pub struct Amd64CoreRegs {
+    // RAX, RBX, RCX, RDX, RSI, RDI, RBP, RSP, r8-r15
+    regs: [u64; 16],
+    // Status register
+    eflags: u32,
+    // Instruction pointer
+    rip: u64,
+    // Segment registers: CS, SS, DS, ES, FS, GS
+    segments: [u32; 6],
+    // FPU registers: ST0 through ST7
+    st_regs: [F80; 8],
+    // SIMD Registers: XMM0 through XMM15
+    xmm_regs: [u128; 0x10],
+    // SSE Status/Control Register
+    mxcsr: u32,
+}
+
+type F80 = [u8; 10];
+
+impl Registers for Amd64CoreRegs {
+    fn gdb_serialize(&self, mut write_byte: impl FnMut(Option<u8>)) {
+        macro_rules! write_bytes {
+            ($bytes:expr) => {
+                for b in $bytes {
+                    write_byte(Some(*b))
+                }
+            };
+        }
+
+        for reg in &self.regs {
+            write_bytes!(&reg.to_le_bytes());
+        }
+
+        // rip
+        write_bytes!(&self.rip.to_le_bytes());
+
+        // eflags
+        write_bytes!(&self.eflags.to_le_bytes());
+
+        // cs, ss, ds, es, fs, gs
+        for seg in &self.segments {
+            write_bytes!(&seg.to_le_bytes());
+        }
+
+        // st0 to st7
+        for st_reg in &self.st_regs {
+            write_bytes!(st_reg);
+        }
+
+        // fctrl, fstat, ftag, fiseg, fioff, foseg, fooff, fop
+        for _ in 0..8 {
+            write_bytes!(&[0; 4]);
+        }
+
+        // xmm0 to xmm15
+        for xmm_reg in &self.xmm_regs {
+            write_bytes!(&xmm_reg.to_le_bytes());
+        }
+
+        // mxcsr
+        write_bytes!(&self.mxcsr.to_le_bytes());
+
+        // padding?
```

Comment:
uhhhh, could you dig into what these bytes actually represent? 😬

Reply:
I did my best (and tried again)... all registers GDB displays with `info all-registers` are accounted for, and no combination of poking at various gdb features, looking at GDB source, and filling that section with various forms of data resulted in me figure out what those bytes are. All I know is that if I include more or less bytes, GDB gets mad. I would love to find out myself, but GDB is not my strongest suit so maybe I'm missing something...

(also, admittedly, I ended up just throwing data at GDB and reverse engineering how it decodes it because I found the x86 target description very hard to correlate into an actual specification of layout)

Reply:
All good, I feel your pain haha. 
GDB is quite the beast, and trying to figure out what's going on can be quite tricky at times.

I'm okay with leaving it in. Maybe add brief `// HACK:` (or `// XXX:`) comment noting your efforts into deciphering these bytes, and that while they must mean _something_, it doesn't seem to be something particularly important.

Additionally, instead of calling `write_bytes!(&[0; 0x18])`, could you replace it with a `(0..0x18).for_each(|_| write_byte(None))` instead? This will serialize as `xxxx...` bytes, indicating that these registers haven't been collected (as opposed to indicating that they've been collected, albeit with a zero value).